hnn_core/pyramidal.py

```diff
@@ -86,8 +116,19 @@ def set_geometry(self, p_dend):
             * cm: membrane capacitance in micro-Farads
             * Ra: axial resistivity in ohm-cm
         """
+        sec_pts, sec_lens, sec_diams, _, topology = self.secs()
+
+        # Connects sections of THIS cell together.
+        for connection in topology:
+            # XXX: risky to use self.soma as default. Unfortunately there isn't
+            # a dictionary with all the sections (including soma)
```

Comment:
Given that `self.create_soma()` is called prior, why is this risky?

Reply:
maybe not risky right now ... but if we loaded a `geo` or `swc` cell from the internet and it somehow was missing a `soma`, it could lead to a crash and a cryptic error. Unlikely but probable? I can remove the comment if you think it's more confusing than helpful

Reply:
Good point. Let's keep the comment for now until we can implement input validation for .geo or .swc cell files.